aws/resource_aws_macie2_invitation_accepter_test.go

```diff
@@ -0,0 +1,167 @@
+	"github.com/hashicorp/terraform-plugin-sdk/v2/terraform"
+	"github.com/terraform-providers/terraform-provider-aws/aws/internal/envvar"
+)
+
+func testAccAwsMacie2InvitationAccepter_basic(t *testing.T) {
+	var providers []*schema.Provider
+	resourceName := "aws_macie2_invitation_accepter.test"
+	email := envvar.TestSkipIfEmpty(t, EnvVarMacie2MemberEmail, EnvVarMacie2MemberEmailMessageError)
+
+	resource.Test(t, resource.TestCase{
+		PreCheck: func() {
+			testAccPreCheck(t)
+			testAccAlternateAccountPreCheck(t)
+		},
+		ProviderFactories: testAccProviderFactoriesAlternate(&providers),
+		CheckDestroy:      testAccCheckAwsMacie2InvitationAccepterDestroy,
+		ErrorCheck:        testAccErrorCheck(t, macie2.EndpointsID),
+		Steps: []resource.TestStep{
+			{
+				Config: testAccAwsMacieInvitationAccepterConfigBasic(email),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckAwsMacie2InvitationAccepterExists(resourceName),
+				),
+			},
+			{
+				Config:            testAccAwsMacieInvitationAccepterConfigBasic(email),
+				ResourceName:      resourceName,
+				ImportState:       true,
+				ImportStateVerify: true,
+			},
+		},
+	})
+}
+
+func testAccCheckAwsMacie2InvitationAccepterExists(resourceName string) resource.TestCheckFunc {
+	return func(s *terraform.State) error {
+		rs, ok := s.RootModule().Resources[resourceName]
+		if !ok {
+			return fmt.Errorf("not found: %s", resourceName)
+		}
+
+		if rs.Primary.ID == "" {
+			return fmt.Errorf("resource (%s) has empty ID", resourceName)
+		}
+
+		conn := testAccProvider.Meta().(*AWSClient).macie2conn
+		input := &macie2.GetAdministratorAccountInput{}
+		output, err := conn.GetAdministratorAccount(input)
+
+		if err != nil {
+			return err
+		}
+
+		if output == nil || output.Administrator == nil || aws.StringValue(output.Administrator.AccountId) == "" {
+			return fmt.Errorf("no administrator account found for: %s", resourceName)
+		}
+
+		return nil
+	}
+}
+
+func testAccCheckAwsMacie2InvitationAccepterDestroy(s *terraform.State) error {
+	conn := testAccProvider.Meta().(*AWSClient).macie2conn
+
+	for _, rs := range s.RootModule().Resources {
+		if rs.Type != "aws_macie2_invitation_accepter" {
+			continue
+		}
+
+		input := &macie2.GetAdministratorAccountInput{}
+		output, err := conn.GetAdministratorAccount(input)
+		if tfawserr.ErrCodeEquals(err, macie2.ErrCodeResourceNotFoundException) ||
+			tfawserr.ErrMessageContains(err, macie2.ErrCodeAccessDeniedException, "Macie is not enabled") {
+			continue
+		}
+
+		if output == nil || output.Administrator == nil || aws.StringValue(output.Administrator.AccountId) != rs.Primary.Attributes["administrator_account_id"] {
+			continue
+		}
+
+		return fmt.Errorf("macie InvitationAccepter %q still exists", rs.Primary.ID)
+
+	}
+
+	return nil
+
+}
+
+func testAccAwsMacieInvitationAccepterConfigBasic(email string) string {
+	return testAccAlternateAccountProviderConfig() + fmt.Sprintf(`
+data "aws_caller_identity" "primary" {
```

Comment:
After thinking about the names, since AWS refers to the inviting account as the "Admin Account", this might be more clear as `admin`